src/Microsoft.DotNet.Cli.Utils/ScriptExecutor.cs

```diff
@@ -76,19 +70,15 @@ public static Command CreateCommandForScript(Project project, string scriptComma
                     // scriptCandidate may be a path relative to the project root. If so, likely will not be found
                     // in the $PATH; add ./ to let bash know where to look.
                     var prefix = Path.IsPathRooted(scriptCandidate) ? string.Empty : "./";
-                    var quote = surroundWithQuotes ? "\"" : string.Empty;
-                    scriptArguments[0] = $"{ quote }{ prefix }{ scriptCandidate }{ quote }";
+                    scriptArguments[0] = $"{prefix}{scriptCandidate}";
                 }
 
                 // Always use /usr/bin/env bash -c in order to support redirection and so on; similar to Windows case.
                 // Unlike Windows, must escape quotation marks within the newly-quoted string.
-                scriptArguments = new[] { "/usr/bin/env", "bash", "-c", "\"" }
-                    .Concat(scriptArguments.Select(argument => argument.Replace("\"", "\\\"")))
-                    .Concat(new[] { "\"" })
-                    .ToArray();
+                scriptArguments = new[] { "/usr/bin/env", "bash", "-c", ArgumentEscaper.EscapeArgArray(scriptArguments) };
             }
 
-            return Command.Create(scriptArguments.FirstOrDefault(), string.Join(" ", scriptArguments.Skip(1)))
+            return Command.Create(scriptArguments.FirstOrDefault(), scriptArguments.Skip(1).ToArray())
```

Comment:
I'm leaving this for now as this is too error prone to change for stabilization, we can file an issue